### ISSUE TITLE
Internet Archive: sync should not try and delete/modify metadata files

### ISSUE THREAD

**Issue by mariomadproductions** [rclone/rclone#6353]
#### The associated forum post URL from `https://forum.rclone.org`
N/A

#### What is the problem you are having with rclone?
If you are creating the item, it works fine, but when the Internet Archive system has created metadata files after upload, and you try and sync again, rclone gives an error as it tries to delete them to match the source directory. These files are not mutable (and shouldn't be), as they are created by the Internet Archive backend automatically.

#### What is your rclone version (output from `rclone version`)
rclone v1.60.0-beta.6375.72227a015

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Linux Mint 20.3 Cinnamon 64-bit

#### Which cloud storage system are you using? (e.g. Google Drive)
Internet Archive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
rclone --progress sync source ia:test3214152142

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
[bug.log](https://github.com/rclone/rclone/files/9234506/bug.log)

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by Nicobuzeta** [comment]
There isn't really anyway for rclone to automatically detect what files you want synced. You should probably setup some filtering (https://rclone.org/filtering/) and set your metadata files (xml and sqlite) to not be synced.

**Comment by ncw** [comment]
> If you are creating the item, it works fine, but when the Internet Archive system has created metadata files after upload, and you try and sync again, rclone gives an error as it tries to delete them to match the source directory. These files are not mutable (and shouldn't be), as they are created by the Internet Archive backend automatically.

What do these file look like? Can you exclude them with a [filter](https://rclone.org/filtering/)?

**Comment by mariomadproductions** [comment]
These are metadata files that exist on every archive.org item, so I think it would make sense to have this as default behaviour, otherwise the sync will always display an error. They have somewhat predictable filenames, but the proper way to detect them appears to be by seeing if their "source" field is set to "metadata" or their format field is set to "Metadata" (one or the other, not both - different metadata files use different ones).

References:
https://archive.org/services/docs/api/metadata-schema/index.html#source
https://archive.org/services/docs/api/metadata-schema/index.html#format
https://help.archive.org/help/files-formats-and-derivatives-a-basic-guide/
[any archive.org item]

**Comment by ncw** [comment]
Is it likely you'd want to download these files?

Maybe we should have a flag which hides them. Perhaps that should be the default?

Thoughts @Lesmiscore ?

**Comment by Lesmiscore** [comment]
It would be nice if you can filter what to sync by metadata value. These files should be filtered out by matching against `source` key (which corresponds to "source" field above).

**Comment by ncw** [comment]
It looks like `source` and `format` appear in rclone metadata

```
$ rclone lsjson -M --stat TestIA:rclone-integration-test/rclone-integration-test_archive.torrent
{
	"Path": "rclone-integration-test_archive.torrent",
	"Name": "rclone-integration-test_archive.torrent",
	"Size": 1846,
	"MimeType": "application/x-bittorrent",
	"ModTime": "2022-08-04T07:53:01.000000000+01:00",
	"IsDir": false,
	"Metadata": {
		"btih": "2f5d9f70dc0172afa946c39d88ed34cdb821b287",
		"crc32": "59bd36d2",
		"format": "Archive BitTorrent",
		"md5": "5a765d1cc5d8f2226836e27937132876",
		"mtime": "2022-08-04T07:53:01+01:00",
		"name": "rclone-integration-test_archive.torrent",
		"rclone-ia-mtime": "1659595981",
		"sha1": "8f0ac5ec6999efe6ad345321b06cfe806b53e99d",
		"size": "1846",
		"source": "metadata"
	}
}
```

So perhaps the best way to attack this would be to add some metadata filtering into the rclone filter system.

So you'd write something like

    rclone sync ....  --metadata-exclude "source=metadata" --metadata-exclude "format=Metadata"

What do you think @Lesmiscore - sensible idea?

**Comment by Lesmiscore** [comment]
Yes, that's what my idea is. It would be _better_ (not necessarily in this case, though) if we can use globs or regex there.

**Comment by mariomadproductions** [comment]
> Is it likely you'd want to download these files?
> 
> Maybe we should have a flag which hides them. Perhaps that should be the default?
> 
> Thoughts @Lesmiscore ?

I think you would want to download them, they contain things like description and tags for the item and hashes for each file etc. The internet archive official command line app downloads them. Its just that they aren't writable (although some fields can be modified indirectly via the IA metadata API), so it doesn't really make sense for rclone to try and write to them, since it will produce an error every time.

**Comment by ncw** [comment]
Can you have a try with this?

[v1.60.0-beta.6390.c520e5269.fix-6353-metadata-filter](https://beta.rclone.org/branch/fix-6353-metadata-filter/v1.60.0-beta.6390.c520e5269.fix-6353-metadata-filter/) on branch [fix-6353-metadata-filter](https://github.com/rclone/rclone/tree/fix-6353-metadata-filter) (uploaded in 15-30 mins)

This implements metadata filtering - this works exactly like the path filtering, but on the metadata written as `key=value`.

So I think for this application `--metadata-exclude "source=metadata" --metadata-exclude "format=Metadata"` will do the trick in `rclone sync`.

      --metadata-exclude stringArray        Exclude metadatas matching pattern
      --metadata-exclude-from stringArray   Read metadata exclude patterns from file (use - to read from stdin)
      --metadata-filter stringArray         Add a metadata filtering rule
      --metadata-filter-from stringArray    Read metadata filtering patterns from a file (use - to read from stdin)
      --metadata-include stringArray        Include metadatas matching pattern
      --metadata-include-from stringArray   Read metadata include patterns from file (use - to read from stdin)

Let me know what you think?

**Comment by mariomadproductions** [comment]
That appears to work - no more errors.
